### PR TITLE
[line tile] set start and end date

### PR DESCRIPTION
**Pull request by chejennifer**
Add handling for setting a start date and end date for data shown in a line tile. This will be used for NL changes to handle a date range query such as "Show me the population of California in the last 5 years"

Screenshot 2023-12-19 at 10 41 12 AM

### CODE REVIEW

**Review by juliawu — COMMENTED** [review]
Thanks for this!

File: static/js/components/tiles/line_tile.tsx [hunk 1]
```diff
@@ -339,6 +343,18 @@ function rawToChart(
       if (obsList.length > 0) {
         const dataPoints: DataPoint[] = [];
         for (const obs of obsList) {
+          if (
+            props.startDate &&
+            obs.date < props.startDate.substring(0, obs.date.length)
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/components/tiles/line_tile.tsx:348]: yes, startDate/endDate is inclusive. I think with this logic, it is inclusive? In the example you gave, 2023 == 2023 so `obs.date < props.startDate.substring(0, obs.date.length)` would be false ..

[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/line_tile.tsx:348]: Oop, should not do late night reviews, you're right! Thanks for responding.

**Review by juliawu — APPROVED** [review]

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**